_posts/2016-05-24-SpringIO16-Spring-Rest-Docs.md

```diff
@@ -0,0 +1,339 @@
+
+
+{% highlight java %}
+private MockMvc mockMvc;
+
+@Autowired
+private WebApplicationContext context;
+
+ @Before
+    public void setup() throws Exception {
+        this.document = document("{method-name}");
+        mockMvc = MockMvcBuilders.webAppContextSetup(wac)
+                .apply(documentationConfiguration(this.restDocumentation).uris().withScheme("https")).alwaysDo(this.document)
+                .addFilter(new JwtFilter(),"/*")
+                .build();
+        objectWriter = objectMapper.writer();
+
+        authToken = TestUtil.getAuthToken();
+        TestUtil.setAuthorities();
+    }
+
+{% endhighlight %}
+
+
+
+#### RestAssured (New 1.1)
+As an alternative, you can use RestAssured to test and document your RESTful services. 
+Available in V1.1, RestAssured will be more expandable than MockMvc.
+
+{% highlight java %}
+private RequestSpecification spec;
+
+@Before
+public void setUp() {
+    this.spec = new RequestSpecBuilder().addFilter(
+            documentationConfiguration(this.restDocumentation)) 
+            .build();
+}
+{% endhighlight %}
+
+
+
+#### Andy's pick
+This time he didn't favor one but he mentioned that RestAssured gives you more functionality and extends your possibilities with HTTP.
+
+
+### Snip, snip, snippets everywhere!
+
+#### Default Snippet
+Snippets are generated by the documented test method.
+Once you run the test method, you can add these snippets in your Markdown/Asciidoctor file. 
+Be aware, these type of snippet will fail if you don't have the correct response/request syntax. 
+
+{% highlight java %}
+this.document.snippets(
+                links(
+                        halLinks(), linkWithRel("self").description("The employee's resource"),
+                        linkWithRel("employee").optional().description("The employee's projection")),
+                        responseFields(
+                                fieldWithPath("username").description("The employee unique database identifier"),
+                                fieldWithPath("firstName").description("The employee's first name"),
+                                fieldWithPath("lastName").description("The employee's last name"),
+                                fieldWithPath("linkedin").description("The employee's linkedin"),
+                                fieldWithPath("unit").description("The employee's unit").type(Unit.class),
+                                fieldWithPath("_links").description("links to other resources")
+                        ));
+ mockMvc.perform(
+                get("/employees/1").accept(MediaType.APPLICATION_JSON));
+
+{% endhighlight %}
+
+
+#### Relaxed Snippet (New in 1.1)
+In contrast to default snippets, relaxed snippets are used to have no complaining when something was neglected in the document.
+This has the advantage if you would like to focus on a certain scenario or specific part of the response/request.
```

Comment:
This is an advantage when you only need to focus on a certain scenario or specific part ...